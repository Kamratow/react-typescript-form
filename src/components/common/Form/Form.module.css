.FormContainer {
    padding: 25px;
    border: 2px solid #000;
}
