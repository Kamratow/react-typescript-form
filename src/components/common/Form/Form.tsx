import React from "react";

import styles from "./Form.module.css";

const Form: React.FunctionComponent = ({ children }) => {
    const handleSubmit = (event: React.FormEvent<HTMLFormElement>): void => {
        event.preventDefault();
    };

    return (
        <form className={styles.FormContainer} onSubmit={handleSubmit}>
            {children}
        </form>
    );
};

export default Form;
