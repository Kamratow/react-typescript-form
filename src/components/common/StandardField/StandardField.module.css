.StandardFieldWrapper {
    margin: 0;
    box-sizing: border-box;
    padding: 5px;
}

.StandardFieldLabel {
    font-size: 20px;
    color: #000;
    font-weight: 600;
    display: block;
    margin-bottom: 8px;
}

.StandardFieldInput {
    font-size: 20px;
    padding: 10px;
    border-radius: 5px;
    width: 100%;
    color: #000;
    border: 1px solid #000;
    box-sizing: border-box;
    margin-bottom: 24px;
}

.StandardFieldInput:focus {
    outline: none;
    box-shadow: 0 0 6pt #008000;
    border-color: #008000;
}

.StandardFieldInputError {
    border-color: rgb(201, 22, 22);
    margin-bottom: 0;
}

.StandardFieldErrorMessage {
    font-size: 16px;
    color: rgb(201, 22, 22);
    margin-top: 3px;
    margin-bottom: 0;
}
