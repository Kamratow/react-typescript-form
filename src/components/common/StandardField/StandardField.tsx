import React, { useState } from "react";

import styles from "./StandardField.module.css";
import { IFormInput } from "../../../models/formInput.model";

export interface IStandardFieldProps extends IFormInput {
    changeHandler: (newValue: string) => void;
}

const StandardField: React.FunctionComponent<IStandardFieldProps> = ({
    id,
    label,
    type,
    value,
    changeHandler,
    errorMessage,
}) => {
    const [isTouched, setIsTouched] = useState<boolean>(false);

    const handleOnChange = (event: React.FormEvent<HTMLInputElement>) => {
        setIsTouched(true);

        changeHandler(event.currentTarget.value);
    };

    const getInputClasses = (): string => {
        const inputClasses: string[] = [];
        inputClasses.push(styles.StandardFieldInput);

        if (errorMessage !== "") {
            inputClasses.push(styles.StandardFieldInputError);
        }

        return inputClasses.join(" ");
    };

    return (
        <div className={styles.StandardFieldWrapper}>
            <label htmlFor={id} className={styles.StandardFieldLabel}>
                {label}
            </label>
            <input
                id={id}
                type={type}
                value={value}
                className={getInputClasses()}
                onChange={handleOnChange}
            />
            {errorMessage !== "" && isTouched && (
                <p
                    id={id + "ErrorMessage"}
                    className={styles.StandardFieldErrorMessage}
                >
                    {errorMessage}
                </p>
            )}
        </div>
    );
};

export default StandardField;
