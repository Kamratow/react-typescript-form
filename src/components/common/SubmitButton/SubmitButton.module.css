.SubmitButtonWrapper {
    margin-top: 30px;
    width: 100%;
}

.SubmitButton {
    display: block;
    border-radius: 15px;
    margin: auto;
    background-color: #000;
    color: #fff;
    cursor: pointer;
    font-size: 25px;
    padding: 15px 25px;
    border: 3px solid #000;
}

.SubmitButton:focus {
    outline: none;
    box-shadow: 0 0 6pt #008000;
    border-color: #008000;
}

.SubmitButton:hover:enabled {
    background-color: #fff;
    color: #000;
}

.SubmitButton:disabled {
    background-color: #a9a9a9;
    cursor: not-allowed;
    border-color: #a9a9a9;
}
