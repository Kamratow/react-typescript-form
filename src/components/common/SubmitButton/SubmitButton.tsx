import React from "react";

import styles from "./SubmitButton.module.css";

export interface ISubmitButtonProps {
    buttonLabel: string;
    disabled: boolean;
}

const SubmitButton: React.FunctionComponent<ISubmitButtonProps> = ({
    buttonLabel,
    disabled,
}) => {
    return (
        <div className={styles.SubmitButtonWrapper}>
            <button
                className={styles.SubmitButton}
                type="submit"
                disabled={disabled}
            >
                {buttonLabel}
            </button>
        </div>
    );
};

export default SubmitButton;
