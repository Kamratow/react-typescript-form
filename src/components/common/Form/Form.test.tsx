import React from "react";
import { shallow } from "enzyme";

import Form from "./Form";
import StandardField from "../StandardField/StandardField";
import SubmitButton from "../SubmitButton/SubmitButton";

describe("Form", () => {
    let mockedChangeHandler: () => {};

    beforeEach(() => {
        mockedChangeHandler = jest.fn();
    });

    it("should render properly with default props", () => {
        const wrapper = shallow(
            <Form>
                <StandardField
                    changeHandler={mockedChangeHandler}
                    errorMessage=""
                    id="testId"
                    type="text"
                    value="test"
                    label="testLabel"
                />
                <SubmitButton
                    buttonLabel="test button label"
                    disabled={false}
                />
            </Form>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it("should properly handle submit", () => {
        const wrapper = shallow(
            <Form>
                <StandardField
                    changeHandler={mockedChangeHandler}
                    errorMessage=""
                    id="testId"
                    type="text"
                    value="test"
                    label="testLabel"
                />
                <SubmitButton
                    buttonLabel="test button label"
                    disabled={false}
                />
            </Form>
        );
        const preventDefaultMock = jest.fn();
        const mockedSubmitEvent = {
            preventDefault: preventDefaultMock,
        };

        const form = wrapper.find("form").at(0);

        expect(preventDefaultMock).toHaveBeenCalledTimes(0);

        form.simulate("submit", mockedSubmitEvent);

        expect(preventDefaultMock).toHaveBeenCalledTimes(1);
    });
});
