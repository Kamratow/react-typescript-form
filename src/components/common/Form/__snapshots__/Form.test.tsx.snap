// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render properly with default props 1`] = `
<form
  className="FormContainer"
  onSubmit={[Function]}
>
  <StandardField
    changeHandler={[MockFunction]}
    errorMessage=""
    id="testId"
    label="testLabel"
    type="text"
    value="test"
  />
  <SubmitButton
    buttonLabel="test button label"
    disabled={false}
  />
</form>
`;
